includes/connect/class-wc-stripe-connect-api.php

```diff
@@ -171,18 +171,44 @@ protected function request( $method, $path, $body = [] ) {
 				'timeout'     => $http_timeout,
 			];
 
-			$args          = apply_filters_deprecated(
+			$args = apply_filters_deprecated(
 				'wc_connect_request_args',
 				[ $args ],
 				'9.6.0',
 				'',
 				'The wc_connect_request_args filter is deprecated since WooCommerce Stripe Gateway 9.6.0, and will be removed in a future version.'
 			);
+
+			if ( WC_Stripe_Helper::is_enhanced_debug_mode_enabled() ) {
+				// Log the request after the filters have been applied.
+				WC_Stripe_Logger::debug(
+					"OAuth: WCC API request: {$method} {$path}",
+					[
+						'current_stripe_api_key' => WC_Stripe_API::get_masked_secret_key(),
+						'url'                    => $url,
+						'headers'                => $headers,
+						'body'                   => WC_Stripe_Connect::redact_sensitive_data( json_decode( $body ) ),
```

Comment:
Do we also want to log the value of `$args`? I am not sure if we do, but it might be helpful.